src/material-experimental/mdc-select/harness/select-harness.spec.ts

```diff
@@ -63,6 +68,11 @@ describe('MatSelectHarness', () => {
 
 /** Shared tests to run on both the original and MDC-based select. */
 function runTests() {
+  afterEach(() => {
+    // Angular won't call this for us so we need to do it ourselves to avoid leaks.
+    overlayContainer.ngOnDestroy();
```

Comment:
We need these changes, because the changes to the container revealed that the tests here are leaking overlay containers. I've fixed it for all harnesses in #17007.